### PR TITLE
Add Color comparison method

### LINKED ISSUE #3645: Color comparison is always false
# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.9.0
            pydantic compiled: True
                 install path: /home/xxxxxxx/xxxxxxx/xxxxxxx/venv310/lib/python3.10/site-packages/pydantic
               python version: 3.10.1 (main, Dec 21 2021, 17:46:38) [GCC 9.3.0]
                     platform: Linux-5.11.0-44-generic-x86_64-with-glibc2.31
     optional deps. installed: ['dotenv', 'typing-extensions']

```

How to reproduce:

```py
>>> from pydantic.color import Color
>>> Color("red") == Color("red")
False
```
As a consequence, when using Color fields, the comparison of equal models is always false.

### PR DESCRIPTION
**Pull request by aminalaee**
## Change Summary

Adds `__eq__` method to `Color` to handle comparisons.

We could also add the comparison method to the `RGBA` class, but I don't see it anyway being used publicly.

## Related issue number

fix #3645

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by aminalaee** [comment]
please review

File: pydantic/color.py [hunk 1]
```diff
@@ -198,6 +198,9 @@ def __str__(self) -> str:
     def __repr_args__(self) -> 'ReprArgs':
         return [(None, self.as_named(fallback=True))] + [('rgb', self.as_rgb_tuple())]  # type: ignore
 
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, Color) and self.original() == other.original()
```
Comment (PrettyWood) [inline pydantic/color.py:202]: Don't we want to check `as_rgb_tuple` ? 🤔
I would personally expect `Color('red') == Color(255, 0, 0)`

Comment (aminalaee) [reply pydantic/color.py:202]: Yeah I wasn't really sure about that, as I posted in the description.

I think that makes sense though. I'll add it to the `RGBA` class too.

**Comment by aminalaee** [comment]
@PrettyWood Doo commits get squashed and merged?

**Comment by PrettyWood** [comment]
Yes

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**